app/models/framework_response/string.rb

```diff
@@ -1,7 +1,7 @@
 class FrameworkResponse
   class String < FrameworkResponse
     validates :value_json, absence: true
-    validates :value_text, on: :update, inclusion: { in: :question_options }, if: :question_options
+    validates :value, inclusion: { in: :question_options }, if: ->(response) { response.question_options && response.value.present? }
```

Comment:
Same as above (What type is object here? I'm wondering if some of this logic expression could be moved to the relevant class and given a method name to make the intent more obvious.)